src/main/java/juke/Juke.java

```diff
@@ -0,0 +1,89 @@
+package juke;
+
+import juke.command.CommandHandler;
+import juke.common.Storage;
+import juke.common.TaskList;
+import juke.common.Ui;
+
+/**
+ * Entry point for the Juke application.
+ */
+public class Juke {
+    private static final Juke INSTANCE = new Juke();
+    
+    private TaskList taskList;
+    private Ui ui;
+    private Storage storage;
+    private boolean hasExited;
+    
+    /**
+     * Constructor that initializes the application.
+     */
+    public Juke() {
+        this.taskList = new TaskList();
+        this.ui = new Ui();
+        this.storage = new Storage(this);
+        this.hasExited = false;
+        CommandHandler.registerCommands();
+    }
+    
+    private void run() {
+        this.ui.greet();
+        this.storage.loadTasks();
+        while (!this.hasExited) {
+            this.ui.runUiLoop();
+        }
+    }
+    
```

Comment:
I like how you abstracted out everything to their constituent classes! Makes code significantly easier to read!